pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/json/MutableJsonIndexImpl.java

```diff
@@ -291,6 +297,88 @@ private RoaringBitmap getMatchingFlattenedDocIds(Predicate predicate) {
       } else {
         return new RoaringBitmap();
       }
+    } else if (predicateType == Predicate.Type.REGEXP_LIKE) {
+      String ceilingKey = _postingListMap.ceilingKey(key + BaseJsonIndexCreator.KEY_VALUE_SEPARATOR);
+      if (ceilingKey == null || !ceilingKey.startsWith(key + BaseJsonIndexCreator.KEY_VALUE_SEPARATOR)) {
+        // No values with this key exist
+        return new RoaringBitmap();
+      }
+
+      Pattern pattern = ((RegexpLikePredicate) predicate).getPattern();
+      MutableRoaringBitmap result = null;
+      char nextChar = BaseJsonIndexCreator.KEY_VALUE_SEPARATOR + 1;
+      SortedMap<String, RoaringBitmap> subMap =
+          _postingListMap.subMap(ceilingKey, true, _postingListMap.floorKey(key + nextChar), true);
```

Comment:
Do we need to calculate floor here? Can we directly put `key + "\u0001"` and make higher bound exclusive? This way we don't need to worry about key ending with "\u0001"

Reply:
`key + "\u0001"` doesn't exist in the map correct? The idea here is to find the key just below `key + "\u0001"` such that it's the that last entry for the key we're searchig.

Reply:
When calling `subMap()`, does it require the lower and upper bound exist in the map? If not, we can simply use `key` as lower bound, `key + '\u0001'` as upper bound, exclusive on both side to get the sub-map

Reply:
Makes sense. subMap does not require the key to be present